sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -539,3 +541,82 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
 
   override def prettyName: String = "str_to_map"
 }
+
+/**
+ * Adds/replaces field in struct by name.
```

Comment:
IIUC, now this expression can only add/replace top-level fields?

Reply:
Sorry I forgot to do this earlier but I've posted a comment now describing the changes I've made and their impact [here](https://github.com/apache/spark/pull/27066#issuecomment-645047815). 

Yes, you are correct. The `WithFields` expression will now only add/replace top-level fields in a struct. It's just a programming exercise to use `WithFields` in conjunction with`GetStructField` to add/replace _nested_ fields in a struct.